lib/matplotlib/colorbar.py

```diff
@@ -854,10 +857,23 @@ def get_ticks(self, minor=False):
             return self._long_axis().get_majorticklocs()
 
     @_api.delete_parameter("3.5", "update_ticks")
-    def set_ticklabels(self, ticklabels, update_ticks=True):
+    def set_ticklabels(self, ticklabels, update_ticks=True, *, minor=False,
+                       **kwargs):
         """
         Set tick labels.
 
+        .. admonition:: Discouraged
+
+            The use of this method is discouraged, because of the dependency
+            on tick positions. In most cases, you'll want to use
+            ``set_ticks(positions, labels=labels)`` instead.
+
+            If you are using this method, you should always fix the tick
+            positions before, e.g. by using `.colorbar.set_ticks` or by
```

Comment:
```suggestion
            positions before, e.g. by using `.Colorbar.set_ticks` or by
```
?

Reply:
Yep, and there was one more mis-reference to axis.set_ticks below.